### DISCUSSION TITLE
질문이 있습니다

### DISCUSSION THREAD

**Discussion by fenoteryn** [bosskmk/pluto_grid#125] [Q&A]
1. row가 클릭되었음을 인지하는 방법이 있나요?
2. Column을 클릭할 때 나오는 freeze ~, autofit을 한글로 바꿀 수 있나요?

**Comment by bosskmk** [comment]
1. https://weblaze.dev/pluto_grid/build/web/#feature/listing-mode
  위의 링크를 참고해 보시면 됩니다.
  mode 를 select 로 실행하면 onSelected 콜백으로 현재 선택된 row 를 탭하거나 엔터키를 입력 했을 때 의 이벤트를 핸들링 할 수 있습니다.

2. configuration 에 localeText 속성으로 `PlutoGridLocaleText.korean()` 을 전달하면 됩니다.

**Reply by fenoteryn** [reply]
configuration에 PlutoGridLocaleText.korean()을 어떻게 전달하는지 예시를 볼 수 있을까요? PlutoGrid 안에 있는 configuration으로 작성해 보니 오류가 뜹니다..

**Comment by bosskmk** [comment]
```dart
// 1
configuration: PlutoGridConfiguration(
  localeText: const PlutoGridLocaleText.korean(),
),

// 2
configuration: PlutoGridConfiguration(
  localeText: const PlutoGridLocaleText(
    freezeColumnToLeft: '좌측 고정',
    freezeColumnToRight: '우측 고정',
  ),
),

// 3
configuration: PlutoGridConfiguration(
  localeText: const PlutoGridLocaleText.korean(
    freezeColumnToLeft: '좌측 고정',
    freezeColumnToRight: '우측 고정',
  ),
),
```

1번은 기본 한글로, 2,3 번은 기본 제공되는 언어에 특정 속성만 변경하고 싶은 문자로 변경 할 수 있습니다.

버전은 1.0.0 이어야 합니다. 0.x 버전에서는 localeText 속성이 없습니다.

**Reply by fenoteryn** [reply]
그럼 0.x 버전에서는 한국어로 바꿀 수 있는 방법이 없는 건가요?

**Reply by bosskmk** [reply]
네, 0.x 는 변경이 안됩니다.

**Reply by fenoteryn** [reply]
grid 안에 있는 데이터 검색하는 기능도 0.x는 지원하지 않나요?

**Reply by bosskmk** [reply]
네~ 1.x 버전으로 업그레이드 하셔서 사용하시는것을 추천드립니다.

업그레이드시 클래스나 속성의 이름이 변경된 것이 대부분이라서 changelog 확인하셔서 오류나는 부분 수정하시면 됩니다.

**Reply by fenoteryn** [reply]
1.x 버전으로 업그레이드를 했더니 column context menu에서 set filter와 reset filter가 생겼는데 이 부분 안 보이게 설정할 수 있는 방법이 있나요? 왼쪽 정렬과 오른쪽 정렬만 보이게 두고 싶습니다.

**Reply by fenoteryn** [reply]
체크되어 있는 테이블의 데이터를 다룰 때, 현재 체크된 row들의 원하는 column value를 받아올 수 있나요?

**Reply by bosskmk** [reply]
> 체크되어 있는 테이블의 데이터를 다룰 때, 현재 체크된 row들의 원하는 column value를 받아올 수 있나요?

https://github.com/bosskmk/pluto_grid/discussions/129#discussioncomment-286592

위 링크를 참고해보십시요.

`rows`, `refRows` 대신에 `checkedRows`, `unCheckedRows` 속성을 사용하면 선택되거나 아닌 행들을 접근 할 수 있습니다.

**Reply by bosskmk** [reply]
> 1.x 버전으로 업그레이드를 했더니 column context menu에서 set filter와 reset filter가 생겼는데 이 부분 안 보이게 설정할 수 있는 방법이 있나요? 왼쪽 정렬과 오른쪽 정렬만 보이게 두고 싶습니다.

https://github.com/bosskmk/pluto_grid/wiki/column-definition-for-korean#column-%EC%86%8D%EC%84%B1-%EB%AA%A9%EB%A1%9D

위 링크를 참고해보십시요.

`enableFilterMenuItem`
`enableHideColumnMenuItem`
`enableSetColumnsMenuItem`

속성의 flag 를 false 로 변경 하면 숨길 수 있습니다.

**Reply by fenoteryn** [reply]
column마다 검색하는 filter 기능이 아닌 grid 전체에서 검색하는 기능을 만들고 싶은데 어떤 방식으로 접근해야 하는지 알 수 있을까요?

**Reply by bosskmk** [reply]
제가 만든 FilteredList 를 반환하는 `stateManager.refRows` 속성에 `setFilter` 를 설정 하고
이후에 `stateManager.notifyListeners()` 를 호출하게 되면 변경 된 필터에 따라 rows 가 변경 될 것 입니다.

https://pub.dev/packages/pluto_filtered_list

**Reply by bosskmk** [reply]
추가로 다른 맥락의 질문은 새로 질문을 등록해서 해주시면 감사하겠습니다.
동일한 문제를 찾는 다른분들이 검색하거나 할 때 유용할 것 입니다.
